9.2/paper.tex

```diff
@@ -703,9 +688,26 @@ \subsection{Advances in GPU support}
 \subsection{Expanded use of C++11 facilities}
 \label{subsec:cxx}
 
-\todo[inline]{Reza: Short paragraph about constexpr}
-
-\todo[inline]{Mention as last sentence that next release will use C++14.}
+Certain types of quantities in a simulation are constants fully known
+at compile time. They can be pre-calculated and fully stored inside the
+compiled program binary in order to avoid unnecessary initialization during
+runtime. This optimization is now enabled for the class templates
+\texttt{Tensor} and \texttt{SymmetricTensor} by qualifying their constructor,
+member functions, and overloaded operators as \texttt{constexpr}.
+For instance, the linear mechanical constitutive model for elastic solids
+uses a constant fourth-order elasticity tensor
+$\mathbb{C} = \lambda \boldsymbol{I} \otimes \boldsymbol{I} + 2 \mu \mathbb{I}$
+which does not depend on the current state of strain.
+This tensor can be statically initialized by defining it as
+\texttt{constexpr SymmetricTensor<4, dim>}.
+As another example, the lattice vectors in a crystal plasticity model are generally
+constant and known during compilation time, enabling their efficient definition as
+\texttt{constexpr Tensor<1, dim>}.
+The capability of defining \texttt{constexpr} variables, functions, and methods
+was introduced by the C++11 standard and was later expanded by the C++14 standard.
+Therefore, the extent of \texttt{constexpr} support in \dealii{} depends on the C++
+standard which is used to compile the library. The next release of \dealii{}
+will fully adopt the features of the C++14 standard.
```

Comment:
... if the compiler allows. We don't enable C++14 constexpr support only conditionally as of now. :slightly_smiling_face:

Reply:
I don't think we should go into details of what will happen in the next release, many details of the the build process might change until the 9.3 (or 10.0) release. I think @bangerth just wanted us to put a small teaser about C++14 support.

Reply:
Yep.